superset-frontend/src/reports/actions/reports.js

```diff
@@ -105,22 +104,10 @@ export const addReport = report => dispatch =>
   SupersetClient.post({
     endpoint: `/api/v1/report/`,
     jsonPayload: report,
-  })
-    .then(({ json }) => {
-      dispatch({ type: ADD_REPORT, json });
-      dispatch(addSuccessToast(t('The report has been created')));
-    })
-    .catch(async e => {
-      const parsedError = await getClientErrorObject(e);
-      const errorMessage = parsedError.message;
-      const errorArr = Object.keys(errorMessage);
-      const error = errorMessage[errorArr[0]];
-      dispatch(
-        addDangerToast(
-          t('An error occurred while editing this report: %s', error),
-        ),
-      );
-    });
+  }).then(({ json }) => {
+    dispatch({ type: ADD_REPORT, json });
+    dispatch(addSuccessToast(t('The report has been created')));
+  });
```

Comment:
What happens when the endpoint returns an error?

Reply:
I moved the catch functionality into the report modal [here](https://github.com/apache/superset/blob/5812a79c0e8eb6cf70297de197534ea9e62a4b52/superset-frontend/src/components/ReportModal/index.tsx#L252-L256), this should handle any errors for this endpoint. Should I also put a catch in the action?

Reply:
Ah, makes sense, because we're no longer showing the error in the toaster, but in the modal. Thanks for clarifying! :)

Reply:
No problem! 😁